apps/yggdrasil/lib/yggdrasil/user.ex

```diff
@@ -9,6 +10,7 @@ defmodule Yggdrasil.User do
     field :hash, :string
     field :password, :string, virtual: true # not part of table
     field :password_confirmation, :string, virtual: true # not part of table
+    belongs_to :role, Yggdrasil.Role
```

Comment:
So I attempted this, more complicated than I felt like doing tonight, will revisit it.

Introducing multiple roles I think is ok, it will introduce a UserRole table, and until ecto 2.0, it will introduce a UserRole model. I did check out the belongs_to as well in this example which will be moot with multiple roles, but it reads that the has_one is used to indicated a one_to_one relationship with the other model having a "belongs_to" relationship. Not sure if that's just an idiom or not though so it may not matter.

The reason for the UserRole is that in order to associate the role to the user in this model the UserRole record has to be created and it's needed for now to do it from what I understand. Could be wrong.

There is technically a form of has_many like the following:
```elixir
has_many :roles, {"users_roles", Yggdrasil.Role}
```
That form is supposed to according to the docs, load Role through the association table, however it seems to be geared toward a one_to_many relationship from the examples in Ecto's test, where for example in Yggdrasil.Role we would have `belongs_to :user, Yggdrasil.User` - which is not what we want this is technically a many to many.

Also has_many is read only, so you can't update through that association. However in, ecto 2.0 with the many_to_many, plus the fact that many_to_many along with has_many, and belongs_to will allow updates through we should be able to eliminate having to model UserRole for example instead we would just be like
```elixir
default_role = # load default... etc..
%User{
  roles: [
    default_role
  ]
}
```
Which will be cleaner, the syntax is something like that. Ecto 2.0 looks really nice I can't wait for that to drop.